src/network/tarcap/packets_handler/handlers/dag_packets_handler.cpp

```diff
@@ -0,0 +1,230 @@
+#include "dag_packets_handler.hpp"
+
+#include "dag/dag_block_manager.hpp"
+#include "network/tarcap/packets_handler/syncing_state.hpp"
+#include "transaction_manager/transaction_manager.hpp"
+
+namespace taraxa::network::tarcap {
+
+DagPacketsHandler::DagPacketsHandler(std::shared_ptr<PeersState> peers_state,
+                                     std::shared_ptr<SyncingState> syncing_state,
+                                     std::shared_ptr<TransactionManager> trx_mgr,
+                                     std::shared_ptr<DagBlockManager> dag_blk_mgr, std::shared_ptr<DbStorage> db,
+                                     uint16_t network_min_dag_block_broadcast, uint16_t network_max_dag_block_broadcast,
+                                     const addr_t &node_addr)
+    : PacketHandler(std::move(peers_state), node_addr, "DAG_BLOCK_PH"),
+      syncing_state_(syncing_state),
```

Comment:
all shared pointer were copied into the ctor so should be moved here into the class members, otherwise we make 1 extra unnecessary copy (I know compiler might optimize it), but lets not rely on complier...